---
layout: post
num: 21
title: Exercise 21
chapter: Chapter 12
description: >
  (2) Demonstrate that a derived-class constructor cannot catch exceptions thrown by its base-class constructor.
---



```java


package com.ciaoshen.thinkinjava.chapter12;

public class Exercise21 {
    public static class BaseException extends Exception {
        private static final long serialVersionUID = 0;
    }
    public static class BaseClass {
        public BaseClass() throws BaseException {
            System.out.println("I am Constructor of BaseClass!");
        }
    }
    public static class DerivedClass extends BaseClass {
        public DerivedClass() {
            try {
                //super();  //ERROR: super() must be the first statement in constructor
            } catch (BaseException e) {
                System.out.println("Exception in BaseClass captured!");
            }
        }
    }
    public static void main(String[] args) {
        DerivedClass derived = new DerivedClass();
    }
}


```



